src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_TabKey.cs

```diff
@@ -67,60 +64,91 @@ private bool TryInvokeSnippetCompletion(TabKeyCommandArgs args, CompletionServic
 
             var text = subjectBuffer.AsTextContainer().CurrentText;
 
-            // Delete the ? and invoke completion
+            // If the user types "<line start><spaces><question><tab>"
```

Comment:
> The editor should also shows the full snippet picker UI in the case of <whatever><identifier><spaces><question><tab>. 

It definitely does not.  Nor would that be a good thing for it to do.  The full snippet picker shows the list of top level folders, but not hte list of items.  As such, you can't use the full snippet picker to type in the name of a snippet and filter down to that item.

```<question><tab>``` brings up the folder oriented list of snippets.

```<id><question><tab>``` brings up the file oriented list of snippets.  Allows you to see all of them and get all our normal completion list filtering and whatnot.

> QuestionMarkIsPrecededByIdentifierAndWhitespace seems weirdly complicated for this task

I agree... but i'm not sure what to replace it with.